packages/button/src/react/__stories__/index.story.tsx

```diff
@@ -214,7 +215,7 @@ export const ExampleAutofocus: Story = () => {
       if (ref && ref.current) ref.current.focus()
     }, [])
 
-    return <Button ref={ref}>Should be focused</Button>
+    return <Button ref={ref as RefFor<'button'>}>Should be focused</Button>
```

Comment:
I felt the same but yea it's just more stuff. I also think we need to go and refactor primitives.ts and types.ts in utils because:
1. the arg pattern is counter to react in some of the Primitives with regards to forwardRef a
2. we need to settle on patterns for uses case not just use types.ts types that obfuscate what we want to to do and we forget how they do it later. We overload so much I found myself running into TS errors about overly complex types throughout this type fix refactoring task.

Reply:
I agree with the "it's just more stuff" aspect of this potential refactor. I'd be grateful if you would form your suggestions for these refactors you mention into new Github issues so that they can be addressed later. Enough detail so they're dev-ready would be great. But if there's a conversation that needs to happen before a decision on approach, staging that conversation would be sufficient. Would you mind doing that, please?

Reply:
On it!